src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMTE_NuclearReactor.java

```diff
@@ -544,87 +524,139 @@ public void explodeMultiblock() {
-		if (spargeGas != null){
-			if (spargeGas.isFluidEqual(Materials.Helium.getGas(1000))){
-				final int outputChances[] = {
-						MathUtils.roundToClosestInt(MathUtils.randInt(10, 1000)/10),
-						MathUtils.roundToClosestInt(MathUtils.randInt(10, 600)/10),
-						MathUtils.roundToClosestInt(MathUtils.randInt(10, 400)/10),
-						MathUtils.roundToClosestInt(MathUtils.randInt(10, 1000)/10),
-						MathUtils.roundToClosestInt(MathUtils.randInt(10, 100)/10)
-				};
-				final int heliumContent = (1000-outputChances[0]-outputChances[1]-outputChances[2]-outputChances[3]-outputChances[4]);
-				Logger.WARNING("Helium remaining: "+heliumContent);
-				outputArrayOfGases = new FluidStack[]{
-						ELEMENT.getInstance().XENON.getFluid(outputChances[0]),
-						ELEMENT.getInstance().NEON.getFluid(outputChances[1]),
-						ELEMENT.getInstance().ARGON.getFluid(outputChances[2]),
-						ELEMENT.getInstance().KRYPTON.getFluid(outputChances[3]),
-						ELEMENT.getInstance().RADON.getFluid(outputChances[4]),
-						Materials.Helium.getGas(heliumContent)
-				};
+	private AutoMap<FluidStack> getByproductsOfSparge(final FluidStack spargeGas){		
+		GasSpargingRecipe aSpargeRecipe = null;		
+		AutoMap<FluidStack> aOutputGases = new AutoMap<FluidStack>();		
+		for (GasSpargingRecipe aRecipe : GasSpargingRecipeMap.mRecipes) {
+			if (aRecipe.mInputGas.isFluidEqual(spargeGas)) {
+				aSpargeRecipe = aRecipe;
 			}
-			else if (spargeGas.isFluidEqual(Materials.Fluorine.getGas(100))){
-				final int outputChances[] = {
-						MathUtils.roundToClosestInt(MathUtils.randDouble(10, 100)),
-						MathUtils.roundToClosestInt(MathUtils.randDouble(1, 50)/10),
-						MathUtils.roundToClosestInt(MathUtils.randDouble(1, 50)/10),
-						MathUtils.roundToClosestInt(MathUtils.randDouble(1, 50)/10)
-				};
-				final int fluorineContent = (100-outputChances[0]-outputChances[1]-outputChances[2]-outputChances[3]);
-				Logger.WARNING("Fluorine remaining: "+fluorineContent);
-				outputArrayOfGases = new FluidStack[]{
-						FLUORIDES.LITHIUM_FLUORIDE.getFluid(outputChances[0]),
-						FLUORIDES.NEPTUNIUM_HEXAFLUORIDE.getFluid(outputChances[1]),
-						FLUORIDES.TECHNETIUM_HEXAFLUORIDE.getFluid(outputChances[2]),
-						FLUORIDES.SELENIUM_HEXAFLUORIDE.getFluid(outputChances[3]),
-						Materials.Fluorine.getGas(fluorineContent)
-				};
+		}
+		if (aSpargeRecipe == null) {
+			return aOutputGases;
+		}		
+		int aSpargeGasAmount = aSpargeRecipe.mInputGas.amount;		
+		FluidStack depletionStack = aSpargeRecipe.mInputGas.copy();
+		depletionStack.amount = aSpargeGasAmount;
+		this.depleteInput(depletionStack);
+		updateSlots();		
+		for (int i=0;i<aSpargeRecipe.mFluidOutputs.length;i++) {
+			if (aSpargeRecipe.mInputGas.isFluidEqual(spargeGas)) {
+				continue; // Skip sparge gas
 			}
+			int aGasAmount = MathUtils.randInt(0, aSpargeRecipe.mMaxOutputQuantity[i]);
+			FluidStack aOutput = aSpargeRecipe.mFluidOutputs[i];
+			aSpargeGasAmount -= aGasAmount;
```

Comment:
Can this ever go negative? Probably need to stop if all input gas is used up.

Reply:
Nope, but this isn't used anymore anyway.
I'll just remove it.

Reply:
Fixed in next push.